src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/params.py

```diff
@@ -10,20 +10,34 @@
     c.register_alias('name', ('--azure-resource-name',))
     c.register_alias('rule_name', ('--name', '-n'))
 
+with ParametersContext(command='monitor alert-rules create') as c:
+    from azure.mgmt.monitor.models.alert_rule_resource import AlertRuleResource
+
+    c.expand('parameters', AlertRuleResource)
+    c.register('condition', ('--condition',),
+               type=json.loads,
+               help='JSON encoded condition configuration. Use @{file} to load from a file.')
+
+for command in ['show', 'delete']:
```

Comment:
I don't appreciate this form of registering command parameters, and it is the very form of registering I was trying to avoid when designing these patterns. The problem with using a loop to register parameters for multiple commands is sacrificing maintainability, readability for merely limited reusability benefits. It makes it difficult to locate where are the parameter registrations are. It requires extra effort when a future change is going to make to only one of the command. It breaks the pattern that parameters associating to one command are cluster together under one indented context. I would advise against this form of parameter registering and prefer duplication.